packages/types/src/datacategory.ts

```diff
@@ -17,4 +17,6 @@ export type DataCategory =
   // Session update events
   | 'session'
   // SDK internal event, like client_reports
-  | 'internal';
+  | 'internal'
+  // Profile event type
+  | 'profile';
```

Comment:
I believe this change is breaking in a very unfortunate way. We use the `DataCategory` type as a callback argument in a hook that can be configured by the users:

https://github.com/getsentry/sentry-javascript/blob/33fd2d6329d48828d177cf867fff4ba4b74c4fb8/packages/types/src/client.ts#L131

We essentially have two options:
- Disregard this fact under the assumption that it won't matter for many people.
- Add logic that prevents `'profile'` from being passed to this hook.

I am fine with both.

Reply:
> Disregard this fact under the assumption that it won't matter for many people

I'm in favor of this, `recordDroppedEvent` is not going to be used by the 99.9% of folks, we can add a note to the changelog about it.

Reply:
@lforst yeah, I figured. As it stands today with the profiling SDK implementation, I dont think recordDroppedEvent will even be called for profiles (I think for now this is fine, but as we integrate more tightly with the SDK, we can fix this)